examples/src/bin/dialog.rs

```diff
@@ -0,0 +1,78 @@
+//! # Dialog Example
+//!
+//! Example of how to obtain the response to a dialog as a future
+
+use gtk::glib::clone;
+use gtk::glib::signal::Inhibit;
+use gtk::prelude::*;
+
+use std::env::args;
+use std::rc::Rc;
+
+async fn dialog<W: IsA<gtk::Window>>(window: Rc<W>) {
+    let question_dialog = gtk::MessageDialogBuilder::new()
+        .transient_for(&*window)
+        .modal(true)
+        .buttons(gtk::ButtonsType::OkCancel)
+        .text("What is your answer?")
+        .build();
+
+    let answer = question_dialog.run_future().await;
+    question_dialog.close();
+
+    let info_dialog = gtk::MessageDialogBuilder::new()
+        .transient_for(&*window)
+        .modal(true)
+        .buttons(gtk::ButtonsType::Close)
+        .text("You answered")
+        .secondary_text(&format!("Your answer: {:?}", answer))
+        .build();
+
+    info_dialog.run_future().await;
```

Comment:
Might make sense to do something with the return value here for example purposes

Reply:
The answer from the previous dialog is shown. I think this is a good example why you might wan't to use async/await even if it's only for closing the dialog afterwards.

Reply:
ok :)